File: NTime/BaseCore/PlayerFilter/PlayerFilterOptions.cs
using BaseCore.DataBase;

namespace BaseCore.PlayerFilter
{
    public class PlayerFilterOptions
    {
        public PlayerSort PlayerSort { get; set; }
        public bool DescendingSort { get; set; }
        public int ExtraDataSortIndex { get; set; }
        public string Query { get; set; }
        public bool? Men { get; set; }
        public bool? WithoutStartTime { get; set; }
        public bool? Invalid { get; set; }
        public bool? CompletedCompetition { get; set; }
        public bool? HasVoid { get; set; }
        public bool? IsPaidUp { get; set; }
        public Distance Distance { get; set; }
        public AgeCategory AgeCategory { get; set; }
        public Subcategory Subcategory { get; set; }
    }
}

File: NTime/Server/ntimeAngular/src/app/Models/Players/PlayerFilterOptions.ts
import { PlayerSort } from '../Enums/PlayerSort';
import { Distance } from '../Distance';
import { AgeCategory } from '../AgeCategory';
import { Subcategory } from '../Subcategory';

export class PlayerFilterOptions {
  PlayerSort: PlayerSort = PlayerSort.ByLastName;
  DescendingSort = false;
  ExtraDataSortIndex: number;
  Query?: string;
  IsMale?: boolean;
  WithoutStartTime?: boolean;
  Invalid?: boolean;
  CompletedCompetition?: boolean;
  HasVoid?: boolean;
  IsPaidUp?: boolean;
  Distance?: Distance;
  AgeCategory?: AgeCategory;
  Subcategory?: Subcategory;
}
